lib/src/cpp/common/LocatorList.hpp

```diff
@@ -0,0 +1,367 @@
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator IP address.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_address(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator unique LAN ID.
+ *        TCPv4 only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCPv4 unique LAN ID.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_unique_lan_id(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator WAN IPv4 address.
+ *        TCPv4 only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCPv4 WAN IP address.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_wan_address(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/************************************************************************/
+/* Query functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Number of locators defined.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ *
+ * @return uint32_t Number of locators in the list.
+ *
+ * @throw ElementNotFound Exception if the list has not been set.
+ */
+uint32_t size(
+        xercesc::DOMElement*& xml_node);
+
+/************************************************************************/
+/* Clear functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Remove locator or complete locator list.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be removed. If empty, the complete list is removed.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+void clear(
+        xercesc::DOMElement*& xml_node,
```

Comment:
In this case, the XML object is modified. But again, the object is not going to be used after this method. Consequently, passing the pointer would be enough. However, it is [safer to pass a reference over a pointer](https://www.geeksforgeeks.org/passing-by-pointer-vs-passing-by-reference-in-cpp/) unless there is a need to reset the variable or to check for a `nullptr`.

```suggestion
        xercesc::DOMElement& xml_node,
```

This suggestion applies to the `setters` as well.